### PR TITLE
Gp 40 migrate binary search tree

### PR DESCRIPTION
**Pull request by maxstasiuk92**
@tboychuk

### CODE REVIEW

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                 * .......\
+                 * ........4
+                 * .........\
+                 * ..........5
+                 */
+                arguments(new Integer[]{1, 2, 3, 4, 5}, 4));
+    }
+
+    @SuppressWarnings("unchecked")
+    private BinarySearchTree<Integer> createTreeWith(Integer... elements) {
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java:112]: @maxstasiuk92 in all exercises except TDD, test classes in `main` and `completed` **should be equal!**
So you'll need to create `RecursiveBinarySearchTree` class in `main`, and throw `ExerciseNotCompletedException` in each method as we do in other exercises in this module. 

The general rule is that **diff between `completed` and `main` should only show an implementation**. Tests, method signatures, java docs, the class location should always be the same in those two branches.

**Review by tboychuk — CHANGES_REQUESTED** [review]

File: 6-0-test-driven-development/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import static org.assertj.core.api.Assertions.fail;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class BinarySearchTreeTest {
+
+    /**
+     * Always failing test, remove it before doing the exercise
+     */
+    @Test
+    @Order(1)
```
Comment (tboychuk) [inline 6-0-test-driven-development/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java:18]: @maxstasiuk92 any improvement is appreciated. And going down the road **please make sure you follow the convention**. It's better when all exercises within a module follow the same structure. For instance, **other TDD exercises don't have this one failing test**. Therefore it's better to either add such a test for every exercise or to none.

**Review by tboychuk — APPROVED** [review]
